src/app/downloads/downloads-chart.component.spec.ts

```diff
@@ -82,17 +86,21 @@ describe('DownloadsChartComponent', () => {
       el = de.nativeElement;
 
       // call episode and podcast metrics to prime the store
-      comp.store.dispatch(castleFilter(filter));
-      comp.store.dispatch(castleEpisodeMetrics(episode, filter, 'downloads', downloads));
-      comp.store.dispatch(castlePodcastMetrics(podcast, filter, 'downloads', downloads));
+      comp.store.dispatch(new CastleFilterAction({filter}));
+      comp.store.dispatch(new CastleEpisodeMetricsAction({episode, filter, metricsType: 'downloads', metrics: epDownloads}));
+      comp.store.dispatch(new CastlePodcastMetricsAction({podcast, filter, metricsType: 'downloads', metrics: podDownloads}));
     });
   }));
 
   it('should transform podcast and episode data to chart models', () => {
-    expect(comp.chartData.length).toBe(2);
+    expect(comp.podcastChartData.data.length).toEqual(podDownloads.length);
     expect(comp.episodeChartData[0].label).toEqual(episode.title);
-    expect(comp.episodeChartData[0].data.length).toEqual(downloads.length);
-    expect(comp.podcastChartData.label).toEqual('All Other Episodes');
-    expect(comp.podcastChartData.data.length).toEqual(downloads.length);
+    expect(comp.episodeChartData[0].data.length).toEqual(epDownloads.length);
+    expect(comp.chartData.length).toBe(2);
+    expect(comp.chartData[comp.chartData.length - 1].label).toEqual('All Other Episodes');
+  });
+
+  it('should subtract episode data from podcast data for chart display', () => {
```

Comment:
I moved the 'All Other Episodes' calculation out of the podcast metrics state. It was messy, and it seemed more like a chart concern.